app/state/reducers/reservationCateringOrdersReducer.js

```diff
@@ -0,0 +1,17 @@
+import keyBy from 'lodash/keyBy';
+import mapValues from 'lodash/mapValues';
+import { handleActions } from 'redux-actions';
+
+import actionTypes from 'api/actionTypes';
+
+// State will map reservation ids to catering order ids.
+const initialState = {};
+
+export default handleActions({
+  [actionTypes.CATERING_ORDER_GET_SUCCESS]: (state, action) => {
+    if (!action.payload.entities.cateringOrders) return state;
+    const orders = Object.values(action.payload.entities.cateringOrders);
+    const newData = mapValues(keyBy(orders, 'reservation'), 'id');
```

Comment:
I think this is a bit difficult to understand.

Maybe: `fromPairs(map(action.payload.entities.cateringOrders, (order) => [order.reservation, order.id]))` has a simpler logic?